### PR TITLE
Update AuthLDAP.php

### PR DESCRIPTION
**Pull request by andreas-bulling**
Fixed errors with PHP 8.1

I encountered multiple errors when running LimeSurvey with PHP 8.1. I fixed these with this pull request.

### CODE REVIEW

File: application/core/plugins/AuthLDAP/AuthLDAP.php [hunk 1]
```diff
@@ -230,8 +230,8 @@ private function _createNewUser($new_user, $password = null)
         if (!is_resource($ldapconn)) {
```
Comment (Shnoulle) [inline application/core/plugins/AuthLDAP/AuthLDAP.php:230]: The issue seems more here :)

Comment (Shnoulle) [reply application/core/plugins/AuthLDAP/AuthLDAP.php:230]: $ldacon return ressource in 7.X, Class in 8.1 and array (with error) on both.

You can just replace here by the same test `if (isset($ldapconn['errorCode'])) {`

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Same test must be done when create new user.

**Comment by Shnoulle** [comment]
@andreas-bulling see https://bugs.limesurvey.org/view.php?id=18427

**Comment by Shnoulle** [comment]
Can i ask you to test https://github.com/LimeSurvey/LimeSurvey/pull/2685

**Comment by andreas-bulling** [comment]
I found this message on the forum, which is related:
https://forums.limesurvey.org/forum/plugins/128491-error-when-authenticating-using-authldap-auto-create-user-fails

**Comment by andreas-bulling** [comment]
Got the latest version from git - back to the error.

As mentioned [here](https://github.com/LimeSurvey/LimeSurvey/pull/2665#issuecomment-1287883764)
authentication in multiple other locations needs to be fixed first before the createNewUser workflow can be tested.

```
500: Internal Server Error
Cannot use object of type LDAP\Connection as array
```

**Comment by Shnoulle** [comment]
Please : https://github.com/LimeSurvey/LimeSurvey/pull/2685

**Comment by andreas-bulling** [comment]
works, including user auto create. Thanks.

**Comment by Shnoulle** [comment]
> works, including user auto create. Thanks.

Thank to you :) 

Can you put this information on Pull request ?

I have PHP81 on some server, but not on the LDAP testing … server …

**Comment by Shnoulle** [comment]
Thank you for testing and view the issue :)

**Pull request closed without merging**